testsuite/tests/weak-ephe-final/finaliser2.ml

```diff
@@ -41,6 +46,7 @@ let[@inline never][@local never] test3 () =
       print_endline "test3: child.finalise") r)
   in
   Domain.join d;
+  Unix.sleep 1;
```

Comment:
Instead of sleeping, could we loop until finalisers are run? I believe that ocamltest supports a timeout option (`timeout = "10"`) to avoid blocking the CI in failure cases.

Reply:
Note: if we want to loop by looking at the value of `!c`, we probably want to make this an `Atomic.t` instead of a reference.